### DISCUSSION TITLE
Markdown Directives

### DISCUSSION THREAD

**Discussion by aFuzzyBear** [withastro/roadmap#142] [Proposal]
# Directives in Markdown

I have been trying to address this use case that I am facing in Markdown, where the `md` compiler does do many wonders for us to write and express content without using html. But there is cases where we do need to use HTML in our markdown documents.

In exploring this matter I came accross the `remark-directive` package, which has merit to it however I had problems implementing. This led me to think that maybe Astro's micro/gold(?)mark complier would be able to with some modification be able to support something similar. 

## Proposed Syntax

Without any knowledge of the AST for the markdown output, one makes the simple and niave assumption that the following can be made possible

```markdown
# This is a Heading
::.title ::#title ::{data-title="This is a heading"} ::aria-labelby="Heading"

Then we can have some text afterwards and apply a class to it
::.para

```

This renders out as
```html
<h1 class="title" id="title" data-title="This is a heading" aria-labelby="Heading"> 
  This is a Heading 
</h1>

<p class="para">
 Then we can have some text afterwards and apply a class to it
</p>
```

Normally for a button we would resort to using `<button class=".btn">Click Me</button>` the alternative would be to draw on the one thing I dont like about the remark-directive plugin is declaring the html elements within the directives.

Now the differenciation is that this is used to provide the html elements additional attribute data to help inform the compiler and then spit out the html with the appropriate html attribute values

So the Idea around this proposal is to have directives used to augment the compiler with additional attribute data not declare html attributes. 

The only downside I see of this is having this in our raw documentation, and having places like github not be able to parse this information out and instead display it on the page. Creating awful documentations in the process.

**Comment by FredKSchott** [comment]
This would be interesting to see as a remark/rehype plugin! That is probably how this would be implemented anyway, if we did bring this into Astro core. However, I think long-term I would recommend that this live as an optional plugin (or in the future, integration) for anyone who wants it, for the downside that you mentioned above.

**Comment by aFuzzyBear** [comment]
Agreed, thank you Fred, just didnt know which would be the better route, Ill take a deeper look into the remark/rehype thing and see if I can make any headwind with that,
